### PR TITLE
Fixes Issue #153. Fixed state bugs. Cleaned up code for Issue #149.

### PR DESCRIPTION
**Pull request by StylusEaterChef**
Signed-off-by: Adam M Dutko <adutko@chef.io>

### Description

Fixed Issue #149 and also cleaned up code with extra documentation.

### Issues Resolved

Issue #149 

### Check List

- [ ] New functionality includes tests
- [X] All tests pass
- [X] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
 source "https://rubygems.org"
 
 gemspec
+gem "test-kitchen"
```
Comment (devopsdina) [inline Gemfile:4]: @StylusEaterChef What requires us to add the test-kitchen gem?

File: README.md [hunk 2]
```diff
@@ -670,11 +670,11 @@ info:    vm image list command OK
...
 
-* The ```destroy_explicit_resource_group_tags``` (default: "true") parameter can be used when you want to remove tags associated with an explicit resource group. The default setting is set to `true` to remain consistent with previous behavior. This should be used in combination with an explicitly named resource group and will be honored during the ```kitchen destroy``` phase.
+* The ```destroy_explicit_resource_group_tags``` (default: "true") parameter can be used when you want to remove tags associated with an explicit resource group. The default setting is set to "true" to remain consistent with previous behavior. This should be used in combination with an explicitly named resource group and will be honored during the ```kitchen destroy``` phase.
```
Comment (devopsdina) [inline README.md:677]: Should we just put ```explicit_resource_group_name``` instead of "an explicitly named resource group"?

File: Gemfile (hunk 1 above)
Comment (StylusEaterChef) [reply Gemfile]: Good question. Not sure how that go in there, removed it. Thanks.

File: README.md (hunk 2 above)
Comment (StylusEaterChef) [reply README.md:677]: I suppose. I hate the usage of "explicit" as it seems weird, but in context it makes sense. I'll reword and adjust.

File: lib/kitchen/driver/azurerm.rb [hunk 3]
```diff
@@ -746,6 +768,26 @@ def get_resource_group
...
       end
 
+      # Checks whether a resource group exists.
+      #
+      # @param resource_group_name [String] The name of the resource group to check.
+      # The name is case insensitive.
+      #
+      # @return [Boolean] operation results.
+      #
+      def resource_group_exists(resource_group_name)
```
Comment (tas50) [inline lib/kitchen/driver/azurerm.rb:778]: If this returns a bool it should end with ?

Comment (StylusEaterChef) [reply lib/kitchen/driver/azurerm.rb:778]: Not sure I understand your suggestion. The function returns the last value and that type is a Boolean type.

Comment (tas50) [reply lib/kitchen/driver/azurerm.rb:778]: Yeah it's just a ruby convention to have true/false methods named with ? so it's obvious when reading the code what would be returned by the method

Comment (StylusEaterChef) [reply lib/kitchen/driver/azurerm.rb]: Changed.

**Pull request merged**